tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp

```diff
@@ -131,13 +131,25 @@ namespace tcf {
             "requesterSignature",
             "invalid request; failed to retrieve requester signature");
 
+        if (data_encryption_algorithm.length() > 0) {
+            tcf::error::ThrowIf<tcf::error::ValueError>(data_encryption_algorithm != "AES-GCM-256",
+                "Unsupported dataEncryptionAlgorithm found in the request");
+        }
+
+        // Convert payload_format to lower case
+        std::transform(payload_format.begin(), payload_format.end(),
+            payload_format.begin(), ::tolower);
+        tcf::error::ThrowIf<tcf::error::ValueError>(payload_format != "json-rpc",
+            "Unsupported payload format found in the input");
+
         // Decrypt Encryption key
         ByteArray encrypted_session_key_bytes = HexStringToBinary(encrypted_session_key);
         session_key = enclaveData.decrypt_message(encrypted_session_key_bytes);
         ByteArray session_key_iv_bytes = HexStringToBinary(session_key_iv);
         JSON_Array* data_array = json_object_get_array(params_object, "inData");
-
         size_t count = json_array_get_count(data_array);
+        tcf::error::ThrowIf<tcf::error::ValueError>(count == 0, "Indata is empty");
```

Comment:
indata can be empty if work order doesn't need any in data

Reply:
As per the spec, it should have one or more items in indata. inData contains either a JWT of the specified data or an array of one or more Work Order inputs, e.g. state, message containing input parameters.